Wikipedia/Code/WelcomeAnimationExtensions.swift

```diff
@@ -0,0 +1,753 @@
+    var finalOpacity: Double
+    var initialTransform: CATransform3D
+    var finalTransform: CATransform3D
+}
+
+extension UIBezierPath {
+    class func wmf_circlePathWithRadius(radius: CGFloat) -> UIBezierPath {
+        return UIBezierPath(
+            arcCenter: CGPointZero,
+            radius: radius,
+            startAngle: 0.0,
+            endAngle: CGFloat(M_PI * 2.0),
+            clockwise: true
+        )
+    }
+    class func wmf_horizontalLinePathWithWidth(width: CGFloat) -> UIBezierPath {
+        let path = UIBezierPath()
+        path.moveToPoint(CGPointMake(width * -0.5, 0.0))
+        path.addLineToPoint(CGPointMake(width * 0.5, 0.0))
+        return path
+    }
+    class func wmf_verticalLinePathWithHeight(height: CGFloat) -> UIBezierPath {
+        let path = UIBezierPath()
+        path.moveToPoint(CGPointMake(0.0, height * -0.5))
+        path.addLineToPoint(CGPointMake(0.0, height * 0.5))
+        return path
+    }
+    class func wmf_plusPathWithWidth(width: CGFloat) -> UIBezierPath {
+        let path = UIBezierPath.wmf_horizontalLinePathWithWidth(width)
+        path.appendPath(UIBezierPath.wmf_verticalLinePathWithHeight(width))
+        return path
+    }
+}
+
+extension CAShapeLayer {
+    func wmf_applyCommonSettingsForSize(size: CGSize, position: CGPoint, initialOpacity: Double) -> Void {
+        self.lineCap = "round"
+        self.opacity = Float(initialOpacity)
+        self.lineWidth = size.width * 0.014
+        self.strokeEnd = 1.0
+        self.position = CGPointMake(size.width * position.x, size.height * position.y)
+        self.zPosition = 100
+    }
+    class func wmf_addLine(line: WelcomeLine, toView: UIView) -> CAShapeLayer {
+        let shape = CAShapeLayer()
+        shape.path = UIBezierPath.wmf_horizontalLinePathWithWidth(toView.frame.size.width * line.width).CGPath
+        shape.strokeColor = UIColor.blackColor().CGColor
+        shape.transform = line.animation.initialTransform
+        shape.wmf_applyCommonSettingsForSize(toView.frame.size, position: line.position, initialOpacity: line.animation.initialOpacity)
+        toView.layer.addSublayer(shape)
+        return shape;
+    }
+    class func wmf_addDashedCircle(circle: WelcomeCircle, toView: UIView) -> CAShapeLayer {
+        let shape = CAShapeLayer()
+        shape.path = UIBezierPath.wmf_circlePathWithRadius(toView.frame.size.width * circle.radius).CGPath
+        shape.fillColor = UIColor.clearColor().CGColor
+        shape.strokeColor = UIColor.blackColor().CGColor
+        shape.lineDashPattern = [toView.frame.size.width * 0.029, toView.frame.size.width * 0.047]
+        shape.transform = circle.animation.initialTransform
+        shape.wmf_applyCommonSettingsForSize(toView.frame.size, position: circle.position, initialOpacity: circle.animation.initialOpacity)
+        toView.layer.addSublayer(shape)
+        return shape;
+    }
+    class func wmf_addCircle(circle: WelcomeCircle, toView: UIView) -> CAShapeLayer {
```

Comment:
Also passing the "owner" of the shape layer is a bit of the inversion. You should return the shape layer so it can be added to a view.